src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -267,9 +267,29 @@ public boolean lessThanTwoOverlappingElements(List<String> list1, List<String> l
 
     /**
      * Return true if there are unbalanced duplicates present
+     * Example 1: A = [1, 2, 3, 4]
+     *            B = [1, 2, 3, 4, 5, 6]
+     *            Balanced duplicated ==> Same order
+     * Example 2: A = [1, 2, 2, 3, 4]
+     *            B = [1, 2 ,3, 4]
+     *            A' = [1, 2, 2, 3, 4]
+     *            B' = [1, 2, 3, 4]
+     *            Unbalanced duplicates ==> Not in the Same order
      * @see 'https://github.com/ga4gh/seqcol-spec/blob/master/docs/decision_record.md#same-order-specification'*/
-    public boolean unbalancedDuplicatesPresent(List<String> list1, List<String> list2) {
-        // TODO: UNDERSTAND THE CONCEPT IN THE SEQCOL-SPECS
+    public boolean unbalancedDuplicatesPresent(List<String> listA, List<String> listB) {
+        List<String> AinB = getCommonFields(listA, listB);
+        List<String> BinA = getCommonFields(listB, listA);
+        AinB.forEach(System.out::print);
+        System.out.println();
+        BinA.forEach(System.out::print);
+        if (AinB.size() != BinA.size()) {
+            return true;
+        }
+        for (int i=0; i<AinB.size(); i++) {
+            if (!AinB.get(i).equals(BinA.get(i))) {
```

Comment:
I actually think this is checking order where it shouldn't, at least if you want this method to just be checking for unbalanced duplicates - for instance, in the example I wrote for your test, I think the method should return `false` as there are no unbalanced duplicates (even though the lists clearly aren't in the same order).

Reply:
Yeah I think I should review the definition to have a better understanding of the unbalanced-duplicates with more test cases. Because it's clearly not the right algorithm.